pkg/operator/controller/dns/controller.go

```diff
@@ -81,6 +83,16 @@ func New(mgr manager.Manager, config Config) (runtimecontroller.Controller, erro
 	if err := c.Watch(source.Kind[client.Object](operatorCache, &iov1.DNSRecord{}, &handler.EnqueueRequestForObject{}, predicate.GenerationChangedPredicate{})); err != nil {
 		return nil, err
 	}
+	// When a DNS record is deleted, there may be a conflicting record that should be published. Watch exclusively for
+	// deletes, and queue a reconcile request for the appropriate conflicting record, if applicable.
+	if err := c.Watch(source.Kind[client.Object](operatorCache, &iov1.DNSRecord{}, handler.EnqueueRequestsFromMapFunc(reconciler.MapOnRecordDelete), predicate.Funcs{
+		CreateFunc:  func(e event.CreateEvent) bool { return false },
+		DeleteFunc:  func(e event.DeleteEvent) bool { return true },
```

Comment:
Add a comment with your findings that the delete event happens when the object is actually deleted, not when it is merely marked for deletion (that is, when `deletionTimestamp` is set).